### ISSUE TITLE
python-xxhash<3 pinning blocks Python 3.11

### ISSUE THREAD

**Issue by danschef** [conda-forge/isofit-feedstock#26]
### Comment:

I tried to create a Python 3.11 environment for EnPT but it seems like the `python-xxhash<3` pinning of isofit blocks it because `python-xxhash<3` is already 2 years old and was never built for Python 3.11:

```bash
>>> mamba search "python-xxhash<3"
python-xxhash                  2.0.2 py310h5eee18b_1  pkgs/main
python-xxhash                  2.0.2 py310h6acc77f_1  conda-forge
python-xxhash                  2.0.2 py310h7f8727e_0  pkgs/main
python-xxhash                  2.0.2 py311h5eee18b_1  pkgs/main
python-xxhash                  2.0.2 py312h5eee18b_1  pkgs/main
python-xxhash                  2.0.2  py36h70b1f00_0  conda-forge
python-xxhash                  2.0.2  py36h7f8727e_0  pkgs/main
python-xxhash                  2.0.2  py36h8f6f2f9_0  conda-forge
python-xxhash                  2.0.2  py37h5e8e339_0  conda-forge
python-xxhash                  2.0.2  py37h5e8e339_1  conda-forge
python-xxhash                  2.0.2  py37h5eee18b_1  pkgs/main
python-xxhash                  2.0.2  py37h6b43d8f_0  conda-forge
python-xxhash                  2.0.2  py37h6b43d8f_1  conda-forge
python-xxhash                  2.0.2  py37h7f8727e_0  pkgs/main
python-xxhash                  2.0.2  py38h497a2fe_0  conda-forge
python-xxhash                  2.0.2  py38h497a2fe_1  conda-forge
python-xxhash                  2.0.2  py38h5eee18b_1  pkgs/main
python-xxhash                  2.0.2  py38h7f8727e_0  pkgs/main
python-xxhash                  2.0.2  py39h3811e60_0  conda-forge
python-xxhash                  2.0.2  py39h3811e60_1  conda-forge
python-xxhash                  2.0.2  py39h5eee18b_1  pkgs/main
python-xxhash                  2.0.2  py39h7f8727e_0  pkgs/main
```

```python
[CIRUNNER__ENPT] 2023/12/12 15:43:58 [I]: #9 [5/7] RUN mamba env create -n ci_env -f /root/environment_enpt.yml &&     conda clean -afy
[CIRUNNER__ENPT] 2023/12/12 15:43:58 [I]: #9 sha256:4763bae284c4d10c29139be0b2493c9f45303da10ee0c4f31c43432f344a54d2
[CIRUNNER__ENPT] 2023/12/12 15:44:15 [I]: #9 16.58
[CIRUNNER__ENPT] 2023/12/12 15:44:15 [I]: #9 16.58
[CIRUNNER__ENPT] 2023/12/12 15:44:15 [I]: #9 16.58 Looking for: ["python[version='>=3.11']", 'pip', "acwater[version='>=0.3.0']", "arosics[version='>1.0.0']", 'cerberus', 'gdal', "geoarray[version='>=0.15.8']", 'jsmin', 'lxml', 'matplotlib', 'mvgavg', 'natsort', 'numpy', 'numpy-indexed', 'packaging', 'pandas', "pyproj[version='>=3.4.0']", "py-tools-ds[version='>=0.14.25']", 'scikit-image', 'scipy', "sensormapgeo[version='>=0.4.0']", "sicor[version='>=0.19.1']", 'tqdm', 'utm', 'flake8', 'pycodestyle', 'pydocstyle', 'pylint', 'pytest', 'pytest-cov', 'sphinx-argparse', 'sphinx_rtd_theme', 'urlchecker']
[CIRUNNER__ENPT] 2023/12/12 15:44:15 [I]: #9 16.58
[CIRUNNER__ENPT] 2023/12/12 15:44:15 [I]: #9 16.58
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 Could not solve for environment specs
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 The following packages are incompatible
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��â�� arosics >1.0.0  is installable with the potential options
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� arosics [1.0.1|1.0.2|...|1.7.0] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��  â��â�� python_abi 3.6.* *_cp36m, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� arosics [1.0.1|1.0.2|...|1.7.8] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��  â��â�� python_abi 3.7.* *_cp37m, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� arosics [1.0.1|1.0.2|...|1.9.4] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��  â��â�� python_abi 3.8.* *_cp38, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� arosics [1.2.4|1.2.5|...|1.9.4] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��  â��â�� python_abi 3.9.* *_cp39, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� arosics [1.7.3|1.7.4|...|1.9.4] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��  â��â�� python_abi 3.10.* *_cp310, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� arosics [1.8.0|1.8.1|...|1.9.4] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��     â��â�� python_abi 3.11.* *_cp311, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��â�� python >=3.11  is installable with the potential options
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� python [3.11.0|3.11.1|...|3.11.6], which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� python 3.12.0 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��  â��â�� python_abi 3.12.* *_cp312, which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��  â��â�� python 3.12.0rc3 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��     â��â�� _python_rc, which does not exist (perhaps a missing channel);
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65 â��â�� sicor >=0.19.1  is installable and it requires
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65    â��â�� isofit >=2.10 , which requires
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65       â��â�� python-xxhash <3  with the potential options
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.0.1|1.3.0|1.4.1|1.4.3] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python [2.7* |>=2.7,<2.8.0a0 ], which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash 1.0.1 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python [3.5* |>=3.5,<3.6.0a0 ], which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash 1.0.1 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python 3.6* , which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.0.1|1.3.0|1.4.1|1.4.3] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python >=3.6,<3.7.0a0 , which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.0.1|1.3.0|1.4.1|1.4.3] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python >=3.7,<3.8.0a0 , which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.4.1|1.4.3] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python >=3.8,<3.9.0a0 , which conflicts with any installable versions previously reported;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash 1.4.3 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 2.7.* *_cp27mu, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.4.3|1.4.4|2.0.0|2.0.2] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 3.6.* *_cp36m, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.4.3|1.4.4|2.0.0|2.0.2] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 3.6 *_pypy36_pp73, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.4.3|1.4.4|2.0.0|2.0.2] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 3.7.* *_cp37m, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [1.4.3|1.4.4|2.0.0|2.0.2] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 3.8.* *_cp38, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash [2.0.0|2.0.2] would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 3.9.* *_cp39, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash 2.0.2 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��  â��â�� python_abi 3.10.* *_cp310, which can be installed;
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65          â��â�� python-xxhash 2.0.2 would require
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 62.65             â��â�� python_abi 3.7 *_pypy37_pp73, which can be installed.
[CIRUNNER__ENPT] 2023/12/12 15:45:01 [I]: #9 ERROR: executor failed running [/bin/bash -c mamba env create -n ci_env -f /root/environment_enpt.yml &&     conda clean -afy]: exit code: 1
```

In the setup.cfg of isofit, `python-xxhash` is just pinned to `>1.2.0` (see [here](https://github.com/isofit/isofit/blob/1818ef6fbdcc88c80e72f3533602410c86ecfd8b/setup.cfg#L39)), so I am wondering if there is a good reason to exclude `python-xxhash>=3` in the isofit conda-forge build.

**Comment by danschef** [comment]
@unbohn Let me know if I should file a PR for this.

**Comment by unbohn** [comment]
@danschef Thanks for bringing this up! Actually no idea where this pinning in the recipe comes from... Should be totally safe to remove that. Please feel free to submit a PR! I'm pretty busy with AGU this week and won't be able to work on this before Monday.

**Comment by danschef** [comment]
See https://github.com/conda-forge/isofit-feedstock/pull/27 (should be merged before #28).
